File: implementations/httpcat-deno/src/logic_test.ts
import { assert, assertEquals } from "@std/assert";

import { getPetCodeImageUrl, PetType } from "./logic.ts";
import { isStatusCodeValid } from "./logic.ts";

Deno.test("getPetCodeImageUrl", () => {
  assertEquals(
    getPetCodeImageUrl(PetType.Cat, 200),
    "https://http.cat/images/200.jpg",
  );
  assertEquals(
    getPetCodeImageUrl(PetType.Dog, 200),
    "https://http.dog/200.jpg",
  );
});

Deno.test("isStatusCodeValid", () => {
  assert(isStatusCodeValid(200));
  assert(!isStatusCodeValid(1));
  assert(!isStatusCodeValid(318));
});

File: implementations/httpcat-kotlin/app/src/test/kotlin/me/denisbourge/httpcat/LogicTest.kt
package me.denisbourge.httpcat

import kotlin.test.Test
import kotlin.test.assertNotNull
import kotlin.test.assertEquals

import java.net.URI

class LogicTest {
    @Test fun getPetCodeImageUrl() {
		var logic = Logic();
		assertEquals(logic.getPetCodeImageUrl(PetType.Cat, 200), URI("https://http.cat/images/200.jpg"));
		assertEquals(logic.getPetCodeImageUrl(PetType.Dog, 200), URI("https://http.dog/200.jpg"));
    }
}
